### PR TITLE
Build libraries with proper Rollup configuration

### PR DESCRIPTION
**Pull request by cedoor**
## Description

This PR fixes some Rollup configurations that were not working properly, adjusting PR #193 output before publishing new versions.

## Related Issue(s)

Re #144

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have run `yarn prettier` and `yarn lint` without getting any errors
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/baby-jubjub/rollup.config.ts [hunk 1]
```diff
@@ -15,24 +17,45 @@ const banner = `/**
...
+                plugins: [terser({ output: { preamble: banner } })]
+            }
+        ],
+        external: [],
+        plugins: [
+            typescript({
+                tsconfig: "./build.tsconfig.json"
+            }),
+            nodeResolve({ preferBuiltins: false, browser: true }),
+            nodePolyfills(),
```
Comment (0xjei) [inline packages/baby-jubjub/rollup.config.ts:57]: Why do we need the polyfills here? Is this a problem related to the `buffer`?

File: packages/eddsa-poseidon/rollup.config.ts [hunk 2]
```diff
@@ -14,19 +16,53 @@ const banner = `/**
...
+const name = pkg.name.split("/")[1].replace(/[-/]./g, (x: string) => x.toUpperCase()[1])
+
+export default [
+    {
+        input: "src/index.ts",
+        output: [
+            { file: pkg.exports["."].require, format: "cjs", banner },
+            { file: pkg.exports["."].default, format: "es", banner }
+        ],
+        external: [
```
Comment (0xjei) [inline packages/eddsa-poseidon/rollup.config.ts:28]: Why are we putting them as external?

File: packages/utils/package.json [hunk 3]
```diff
@@ -48,16 +48,6 @@
             "types": "./dist/types/f1-field.d.ts",
             "require": "./dist/lib.commonjs/f1-field.cjs",
             "default": "./dist/lib.esm/f1-field.js"
-        },
-        "./crypto": {
```
Comment (0xjei) [inline packages/utils/package.json:52]: why are we removing this?

**Review by 0xjei — APPROVED** [review]
Works fine, thank you for improving the build process. Just left some comments to better understand the configs but ready to merge

File: packages/baby-jubjub/rollup.config.ts (hunk 1 above)
Comment (cedoor) [reply packages/baby-jubjub/rollup.config.ts:57]: Yes, it looks like Rollup still wants the polyfill for `buffer`, even if we're using a package (from which polyfill is based I think). I've tried with `commonjs` and `node-resolve` but they're not enough.

File: packages/eddsa-poseidon/rollup.config.ts (hunk 2 above)
Comment (cedoor) [reply packages/eddsa-poseidon/rollup.config.ts:28]: Because we don't want them to be included in the final bundle. They should be external dependencies.

File: packages/utils/package.json (hunk 3 above)
Comment (cedoor) [reply packages/utils/package.json:52]: The Rollup conf with `preserveModules: true`, which is used to compile the ts files to js without bundling in 1 file, doesn't include the `crypto/crypto.browser.ts` file, because it is not actually imported by `index.ts`. I haven't found any simple solution so far, so I opted not to use that conditional export for now. `crypto` can be imported from `@zk-kit/utils` only for new.
